pandas/tests/dtypes/test_inference.py

```diff
@@ -462,6 +463,12 @@ def test_floats(self):
         result = lib.infer_dtype(arr)
         assert result == 'floating'
 
+    def test_decimals(self):
+        # GH15690
+        arr = np.array([Decimal(1), Decimal(2), Decimal(3)])
+        result = lib.infer_dtype(arr)
+        assert result == 'decimal'
+
```

Comment:
Sorry, could you add a test with a mix of Decimal and non-decimal (like floats) and make sure that the return is `mixed`? Or search through and see if we have a test that already covers this. Other than that, this looks great.